### PR TITLE
Adding a admin context check used for the management calls

### PR DESCRIPTION
**Pull request by rnirmal**
This is a stop gap solution for us the verify a user with admin privileges. It would be nice to reuse the nova policy stuff or have something more concrete. The nova policy stuff is in openstack-common, we just don't have it updated yet.

### CODE REVIEW

**Comment by hub-cap** [comment]
@rnirmal how come you didnt modify the preexisting authorize method? https://github.com/hub-cap/reddwarf_lite/blob/master/reddwarf/common/auth.py#L60

File: reddwarf/common/wsgi.py [hunk 1]
```diff
@@ -315,10 +317,17 @@ def process_request(self, request):
...
+        is_admin = False
+        for role in roles:
+            if role.lower() in self.admin_roles:
+                is_admin = True
+                break
         limits = self._extract_limits(request.params)
         context = rd_context.ReddwarfContext(auth_tok=auth_tok,
                                              tenant=tenant_id,
                                              user=user,
+                                             is_admin = is_admin,
```
Comment (hub-cap) [inline reddwarf/common/wsgi.py]: remove spaces

**Comment by rnirmal** [comment]
@hub-cap I must have dozed off while adding the second authorize right below the existing one :). The Tenant based auth seems to be like general authorize for all calls, not specifically admin calls. We should be able to modify that, if we know a list of calls that don't require admin stored somewhere, since it's in the pipeline before the reddwarf app.

**Comment by rnirmal** [comment]
this one is meant to be used a decorator to the admin required calls, btw changed authorize to admin_context to differentiate it. Let me know if you think of other options to do this.

@auth.admin_context

File: reddwarf/common/wsgi.py (hunk 1 above)
Comment (rnirmal) [inline reddwarf/common/wsgi.py]: fixed

**Comment by imsplitbit** [comment]
+1

**Comment by TimSimpsonR** [comment]
+1

File: reddwarf/common/wsgi.py [hunk 2]
```diff
@@ -305,6 +306,7 @@ def __call__(self, req):
 class ContextMiddleware(openstack_wsgi.Middleware):
 
     def __init__(self, application):
+        self.admin_roles = config.Config.get('admin_roles', [])
```
Comment (cp16net) [inline reddwarf/common/wsgi.py:309]: will this load the list correctly? or will it read in the value just as a string and need parsing to make it a list? please verify this.

Comment (rnirmal) [inline reddwarf/common/wsgi.py:309]: That was verified before I did the PR :) it works(tm) with a single entry or multiple entries.

Comment (cp16net) [inline reddwarf/common/wsgi.py:309]: awesome!

**Comment by cp16net** [comment]
+1

**Comment by hub-cap** [comment]
+1

**Comment by ed-** [comment]
+1

**Pull request merged**